stl/inc/deque

```diff
@@ -1466,7 +1466,7 @@ private:
             _Uninitialized_value_construct_n_unchecked1(_Myptr, _Count); // clear rest to initial block
         }
 
-        _Destroy_range(_Map() + _Myboff, _Map() + _Mapsize());
+        _Destroy_range(_Map(), _Map() + _Mapsize());

```

Comment:
I find it super dubious that we always destroy the objects in the range but only sometimes deallocate the range.

That said the whole implementation is a bit special so ....

Reply:
I think the range is deallocate if and only if `_Map()` is not null. And when `_Map()` is null, the `_Destroy_range` call is no-op.

Reply:
Yes, but then why not pull it into the if?

Reply:
Nice. I'm doing this now.